Google.PowerShell/Compute/GceCmdlet.cs

```diff
@@ -110,6 +122,51 @@ private void WriteWarnings(Operation op)
             }
         }
 
+        private void WriteProgress(Operation op, string progressMessage)
+        {
+            int activityId = (int)(op.Id % int.MaxValue ?? 0);
+            string activity = progressMessage ?? op.Description;
+            if (activity == null)
+            {
+                string operationType = op.OperationType;
+                operationType = operationType.First().ToString().ToUpper() + operationType.Substring(1);
+                int startIndex = op.TargetLink.IndexOf("projects", StringComparison.Ordinal);
```

Comment:
There is far too much magic going on here. We cannot assume that "projects" is part of the internal naming mechanism of the operations. To do so would be taking a dependency on an internal implementation detail.

Instead, we should just provide a lame, generic message like "processing". If it is lame, then the cmdlet author should wire in a more meaningful name.